smt_proofs_test.go

```diff
@@ -234,20 +233,33 @@ func TestSMT_ProveClosestEmptyAndOneNode(t *testing.T) {
 	path := sha256.Sum256([]byte("testKey2"))
 	flipPathBit(path[:], 3)
 	flipPathBit(path[:], 6)
-	closestPath, closestValueHash, proof, err = smt.ProveClosest(path[:])
+	proof, err = smt.ProveClosest(path[:])
 	require.NoError(t, err)
-	require.Equal(t, proof, &SparseMerkleProof{})
-
-	result := VerifyProof(proof, smt.Root(), closestPath, closestValueHash, NoPrehashSpec(sha256.New(), false))
+	require.Equal(t, proof, &SparseMerkleClosestProof{
+		Path:         path[:],
+		FlippedBits:  []int{0},
+		Depth:        0,
+		ClosestPath:  placeholder(smt.Spec()),
+		ClosestProof: &SparseMerkleProof{},
+	})
+
+	result := VerifyClosestProof(proof, smt.Root(), NoPrehashSpec(sha256.New(), false))
 	require.True(t, result)
 
 	require.NoError(t, smt.Update([]byte("foo"), []byte("bar")))
-	closestPath, closestValueHash, proof, err = smt.ProveClosest(path[:])
+	proof, err = smt.ProveClosest(path[:])
 	require.NoError(t, err)
-	require.Equal(t, proof, &SparseMerkleProof{})
-	require.Equal(t, closestValueHash, []byte("bar"))
-
-	result = VerifyProof(proof, smt.Root(), closestPath, closestValueHash, NoPrehashSpec(sha256.New(), false))
+	closestPath := sha256.Sum256([]byte("foo"))
+	require.Equal(t, proof, &SparseMerkleClosestProof{
+		Path:             path[:],
+		FlippedBits:      []int{},
+		Depth:            0,
+		ClosestPath:      closestPath[:],
+		ClosestValueHash: []byte("bar"),
+		ClosestProof:     &SparseMerkleProof{},
+	})
+
+	result = VerifyClosestProof(proof, smt.Root(), NoPrehashSpec(sha256.New(), false))
 	require.True(t, result)
```

Comment:
Can you add a test where we flip bits for very large heights (with more leaves) to verify that it still works to relive the concern I had in one of the other comments.